cmd/eskip/readclient.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/eskipfile"
+	etcdclient "github.com/zalando/skipper/etcd"
+	innkeeperclient "github.com/zalando/skipper/innkeeper"
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+type ReadClient interface {
+	LoadAndParseAll() ([]*eskip.RouteInfo, error)
+
+	LoadUpdate() (eskip.RouteList, []string, error)
+}
+
+type stdinReader struct {
+	reader io.Reader
+}
+
+type inlineReader struct {
+	routes string
+}
+
+type idsReader struct {
+	ids []string
+}
+
+// TODO eliminate duplicate code in write and read clients
+func createReadClient(m *medium) (ReadClient, error) {
```

Comment:
i deeply disagree with most of this in the context of this particular code.
- the dependency inversion principle shouldn't be about some simple functions, but more complex structures. It looks like you are introducing here unnecessary dependencies, just to have an excuse to apply a principle that is unnecessary in this case.
- you write "separation of concerns" is "important when building complex systems". That's right, but for one, this tool is not a complex system. For second, the explanation of the example sounds wrong to me because the client creation concern is introduced for many of the cases where this concern was not there before (for more cases than there was a client), also because the readability is not improved (just read this code), as while the size of the load file got smaller, the total size of the code got much bigger, also because it is not true that you don't have to understand what everything happens in the load file to add a reader, one shouldn't add a reader without understanding something as simple what happens in the load file, and also because the the testability hasn't improved by being able to mock the read client, because the read client wasn't there to start with and there was no need to mock it.
- single responsibility principle: creating a client is not another concern when there is no need to create a client
- i am looking at load.go in master and i don't get the last two comments from the reply. Which abstractions do you mean? Why the files? The load.go in master has 181 lines including the license notice, and the longest function has 21 lines. Aren't we overkilling here something with the ReadClient?